### DISCUSSION TITLE
在uniapp中，使用pnpm安装alova运行时报错

### DISCUSSION THREAD

**Discussion by hormers** [alovajs/alova#535] [Q&A] [answered]
如图，[uniapp-vue3](https://gitee.com/h_mo/uniapp-vue3-vite-ts-template)项目使用pnpm安装alova，运行后会报错，使用yarn或npm不会出现此错误，我不知道这是不是bug，还是使用的方法不对，alova v3版本一直有这个问题，当然这可以用yarn或npm解决，但难免出现必须用pnpm的情况，比如使用了pnpm 的 workspace 功能，无法替换其他包管理工具
![image](https://github.com/user-attachments/assets/9da7549e-e6ad-4a8f-bbec-8d14d3034be1)

**Comment by MeetinaXD** [comment, marked as answer]
在 `uniapp` 项目中，可以先使用 `@rollup/plugin-node-resolve` plugin 解决。

![image](https://github.com/user-attachments/assets/6f33ed99-736d-4193-bd13-c6a2e285fa4d)

**Reply by JOU-amjs** [reply]
还有另一种更方便的方法是，直接再安装下`@alova/shared`，不需要再额外配置。

**Comment by wxfeiang** [comment]
uniApp 的同时升级 alova  @alova/shared  @alova/adapter-uniapp
